### DISCUSSION TITLE
Minimal Epic Stack

### DISCUSSION THREAD

**Discussion by klasco** [epicweb-dev/epic-stack#157] [Ideas]
Hello, I am new to remix and would love to make some projects, but we mostly use Postgres and self-host our apps so if it's possible can I suggest making a sister stack in which we can choose our deployment env, server. Cause there is lot of caching going on which only works for SQLite. Like a minimal stack which everyone needs without batteries includes.

**Comment by sethdavis512** [comment]
Have you checked out some of the other Remix Stacks on GitHub? (https://github.com/topics/remix-stack) I'd imagine there's something that might fit your use case or you could fork something that's close to what you want and make it your own...food for thought.

**Reply by klasco** [reply]
I was just saying that while we are making something so unique and helpfull with all the auth, server timings, cache, monitoring, etc y not have another stack that covers other type of audience.

Aren't there like 20 diff stacks with sqlite that deploy to fly so y can't we make something for the other halves but it depends how you see it.

**Reply by klasco** [reply]
Like tbh I can't get the esm right with hmr and then you have to go through all the auth with totp stuff and the list goes on I making something (junior dev) won't be something as let's say Kent making it experience matters a lot.

**Comment by kentcdodds** [comment]
Hi @klasco,

I definitely understand the desire here, but this would distract from the primary goals and guiding principles of the project. The idea is that I select things I'm sure will work for the kinds of apps I build and people take it or leave it. Switching to postgres should only take a few minutes after bootstrapping the app thanks to Prisma.

But I'm holding firm on the guiding principle of only one way to do things and limiting choices people have to make to be productive. Even if you're used to postgres, I'm confident SQLite will work just fine for you. Feel free to read the decision document about SQLite here: https://github.com/epicweb-dev/epic-stack/blob/main/docs/decisions/003-sqlite.md

Thanks!

**Reply by cr101** [reply]
@kentcdodds I need to use Full-text search for my web app but Prisma Client only supports full-text search for PostgreSQL & MySql databases.

**Reply by kentcdodds** [reply]
For search or complex queries not supported by Prisma, I used better-sqlite3 which is already installed. I've not tried this yet, but for full text search you could try https://sqlite.org/fts5.html

**Reply by cr101** [reply]
The issue with the SqLite FTS5 is that it use a virtual table that doesn't support types, constraints or PRIMARY KEY declarations.

**Reply by kentcdodds** [reply]
Like I said, I haven't tried it yet, but I would suggest diving into this and seeing how critical those issues are to your use case. If they're very important then you could switch to Postgres.
